app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java

```diff
@@ -215,7 +247,8 @@ private void addNewDevice(final Intent intent) {
         final BluetoothDevice device = intent.getParcelableExtra(BluetoothDevice.EXTRA_DEVICE);
 
         // Skip paired devices and devices without a name.
-        if (device.getBondState() != BluetoothDevice.BOND_BONDED && !device.getName().equals("null")) {
+        if (device != null && device.getName() != null
+            && device.getBondState() != BluetoothDevice.BOND_BONDED && !device.getName().equals("null")) {
```

Comment:
I agree that there are a few null checks but everything verbose enough to me to make it clear. The new metgod would be so small and we already have a pmd warnibg about that.